inference-engine/src/transformations/src/transformations/common_optimizations/pad_fusion.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/pad_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset5.hpp>
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(pass::PadFusion, "PadFusion", 0);
+
+template <typename T>
+static bool can_be_fused(const std::shared_ptr<opset5::Pad>& pad, const std::shared_ptr<T>& node,
+                         const std::shared_ptr<opset5::Constant>& pad_value_const,
+                         const std::shared_ptr<opset5::Constant>& pads_begin, const std::shared_ptr<opset5::Constant>& pads_end) {
+    if (!pad)
+        return false;
+    if (!node)
+        return false;
+    if (pad->get_pad_mode() != op::PadMode::CONSTANT)
+        return false;
+
+    if (!pad_value_const)
+        return false;
```

Comment:
Can we fuse `if`s?

Reply:
I can fuse lines 23 & 27. I think fusing other will be less readable.